src/entities/swapRefactorDraft.ts

```diff
@@ -0,0 +1,376 @@
+import { TokenAmount } from './tokenAmount';
+import { SingleSwap, SwapKind, BatchSwapStep } from '../types';
+import {
+    abs,
+    BALANCER_QUERIES,
+    DEFAULT_USERDATA,
+    DEFAULT_FUND_MANAGMENT,
+    ZERO_ADDRESS,
+    NATIVE_ADDRESS,
+    MathSol,
+} from '../utils';
+import {
+    Address,
+    createPublicClient,
+    encodeFunctionData,
+    getContract,
+    http,
+} from 'viem';
+import { balancerQueriesAbi } from '../abi';
+import { PriceImpactAmount } from './priceImpactAmount';
+import cloneDeep from 'lodash.clonedeep';
+import { Token } from './token';
+import { MinimalToken } from '..';
+
+/**
+API Returns list of GqlSorPath:
+
+type GqlSorPath {
+   pools: [String]! --- note can this be address?
```

Comment:
Considering this needs to support both v2 and v3, I believe we'll have to stick with id instead of address. For v3 api will copy pool address into pool id, so we should be covered.